frontend/src/pages/error/error.page.tsx

```diff
@@ -0,0 +1,45 @@
+import { ButtonGroup, Heading, Text } from '@chakra-ui/react'
+import { useEffect, useState } from 'react'
+import { Helmet } from 'react-helmet'
+import { Navigate } from 'react-router-dom'
+import { MessageTypes, useServiceContext } from '../../api/contexts/service/ServiceContext'
+import { LinkButton } from '../../common-components/LinkButton'
+import { CmschPage } from '../../common-components/layout/CmschPage'
+import { UnauthorizedPage } from './unauthorized.page'
+
+type Props = {
+  message?: string
+}
+
+export const ErrorPage = ({ message: messageProp }: Props) => {
+  const { clearMessage, message, type } = useServiceContext()
+  const [clonedMessage, setClonedMessage] = useState<string | undefined>('')
+  const [clonedMessageType, setClonedMessageType] = useState<MessageTypes>(MessageTypes.GENERAL)
+  useEffect(() => {
+    // Cloning the error is needed to clear the error globally
+    // The message from prop can override the message
+    setClonedMessage(messageProp || message)
+    setClonedMessageType(type || MessageTypes.GENERAL)
+    // Clear the error from the context since the user has already been notified, and prepare for navigation
+    clearMessage()
+    // No need for deps because the clearError would clear this error page too
+  }, [])
+  // If there is no error ATM, redirect to home page
+  if (clonedMessage === undefined) return <Navigate to="/" />
+  // Dispaly authentication page for the corresponding error type
+  if (clonedMessageType === MessageTypes.AUTHENTICATION) return <UnauthorizedPage />
```

Comment:
Well I don't like this shitstorm of a code with that many comments. But it can stay as is. 

Just remember: react-router-dom supports navigation states. So you can pass states when navigating, like these:

```ts
navigate('/', { state: { errorMsgs: ['lol', 'lmao'], title: 'kitties died', backPath: -1 } })
``` 

```tsx
return <Navigate replace to="/" state={{ errorMsgs: ['lol', 'lmao'], title: 'kitties died', backPath: -1 }} />`
```

(`backPath: -1` means it goes back by one history entry. The type of backPath could be `To` from the react-router-dom package.)

It could replace the error message handling in the service context entirely. But it's a big refactor, don't need it now.

Reply:
I was just sharing my experiences, not to force anything